lib/cc/engine/analyzers/violation.rb

```diff
@@ -57,7 +61,15 @@ def name
         end
 
         def calculate_points
```

Comment:
This new implementation seems involved enough to be worth speccing. Seems like it might be time to create a spec for this class.